src/cpp/filesystem.hpp

```diff
@@ -0,0 +1,73 @@
+// Define filesystem library
+// Copied from https://stackoverflow.com/a/53365539
+
```

Comment:
This file figures out which version of the filesystem library to include.

Reply:
Okay sounds good.